app/billing/rest.py

```diff
@@ -117,14 +117,15 @@ def get_free_sms_fragment_limit(service_id):
                 financial_year_start = get_current_financial_year_start_year()
 
             if int(financial_year_start) < get_current_financial_year_start_year():
-                result = sms_list[0]   # The oldest entry
+                annual_billing = sms_list[0]   # The oldest entry
             else:
-                result = sms_list[-1]  # The newest entry
+                annual_billing = sms_list[-1]  # The newest entry
```

Comment:
This could be `latest_free_fragment_limit = sms_list[-1].free_sms_fragment_limit` and then you can pass in this value to the `dao_create_or_update_annual_billing_for_year`, this would make the code here much clearer.

As it looks like you are overwriting the `annual_billing` variable after setting it.

Reply:
Think this is OK as no chance with both exist. It's either past or future